webapp/components/Modal/DeleteModal.spec.js

```diff
@@ -0,0 +1,155 @@
+import { shallowMount, mount, createLocalVue } from '@vue/test-utils'
+import DeleteModal from './DeleteModal.vue'
+import Vue from 'vue'
+import Vuex from 'vuex'
+import Styleguide from '@human-connection/styleguide'
+import VueRouter from 'vue-router'
+
+const routes = [{ path: '/' }]
+const router = new VueRouter({ routes })
+const localVue = createLocalVue()
+
+localVue.use(Vuex)
+localVue.use(Styleguide)
+localVue.use(VueRouter)
+
+describe('DeleteModal.vue', () => {
+  let wrapper
+  let Wrapper
+  let propsData
+  let mocks
+
+  beforeEach(() => {
+    propsData = {
+      type: 'contribution',
+      id: 'c300'
+    }
+    mocks = {
+      $t: jest.fn(),
+      $filters: {
+        truncate: a => a
+      },
+      $toast: {
+        success: () => {},
+        error: () => {}
+      },
+      $apollo: {
+        mutate: jest.fn().mockResolvedValue()
+      }
+    }
+  })
+
+  describe('shallowMount', () => {
+    const Wrapper = () => {
+      return shallowMount(DeleteModal, { propsData, mocks, localVue, router })
+    }
+
+    describe('defaults', () => {
+      it('success false', () => {
+        expect(Wrapper().vm.success).toBe(false)
+      })
+
+      it('loading false', () => {
+        expect(Wrapper().vm.loading).toBe(false)
+      })
+    })
+
+    describe('given a post', () => {
+      beforeEach(() => {
+        propsData = {
+          id: 'p23',
+          type: 'post',
+          name: 'It is a post'
+        }
+      })
+
+      it('mentions post title', () => {
+        Wrapper()
+        const calls = mocks.$t.mock.calls
+        const expected = [['post.delete.message', { name: 'It is a post' }]]
+        expect(calls).toEqual(expect.arrayContaining(expected))
+      })
+    })
+  })
+
+  describe('mount', () => {
+    const Wrapper = () => {
+      return mount(DeleteModal, { propsData, mocks, localVue, router })
+    }
+
+    beforeEach(jest.useFakeTimers)
+
+    it('renders', () => {
+      expect(Wrapper().is('div')).toBe(true)
+    })
+
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-05-08T21:44:02Z" title="Wednesday, May 8th 2019, 11:44:02 pm +02:00">May 8, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -0,0 +1,155 @@
+import { shallowMount, mount, createLocalVue } from '@vue/test-utils'
+import DeleteModal from './DeleteModal.vue'
+import Vue from 'vue'
+import Vuex from 'vuex'
+import Styleguide from '@human-connection/styleguide'
+import VueRouter from 'vue-router'
+
+const routes = [{ path: '/' }]
+const router = new VueRouter({routes})
+const localVue = createLocalVue()
+
+localVue.use(Vuex)
+localVue.use(Styleguide)
+localVue.use(VueRouter)
+
+describe('DeleteModal.vue', () => {
+  let wrapper
+  let Wrapper
+  let propsData
+  let mocks
+
+  beforeEach(() => {
+    propsData = {
+      type: 'contribution',
+      id: 'c300'
+    }
+    mocks = {
+      $t: jest.fn(),
+      $filters: {
+        truncate: a => a
+      },
+      $toast: {
+        success: () => {},
+        error: () => {}
+      },
+      $apollo: {
+        mutate: jest.fn().mockResolvedValue()
+      }
+    }
+  })
+
+  describe('shallowMount', () => {
+    const Wrapper = () => {
+      return shallowMount(DeleteModal, { propsData, mocks, localVue, router })
+    }
+
+    describe('defaults', () => {
+      it('success false', () => {
+        expect(Wrapper().vm.success).toBe(false)
+      })
+
+      it('loading false', () => {
+        expect(Wrapper().vm.loading).toBe(false)
+      })
+    })
+    
+    describe('given a post', () => {
+      beforeEach(() => {
+        propsData = {
+          id: 'p23',
+          type: 'post',
+          name: 'It is a post'
+        }
+      })
+
+      it('mentions post title', () => {
+        Wrapper()
+        const calls = mocks.$t.mock.calls
+        const expected = [['post.delete.message', { name: 'It is a post' }]]
+        expect(calls).toEqual(expect.arrayContaining(expected))
+      })
+    })
+  })
+
+  describe('mount', () => {
+    const Wrapper = () => {
+      return mount(DeleteModal, { propsData, mocks, localVue, router})
+    }
+
+    beforeEach(jest.useFakeTimers)
+
+    it('renders', () => {
+      expect(Wrapper().is('div')).toBe(true)
+    })
+
```


I know there are many examples of this in the code base, but I am not a fan of these `renders` tests
I think we could open a ticket to remove them @roschaefer, to be honest. 
If the component does not render, then all the other tests will fail and therefore are testing that the component renders already.
It might not take a long time to run these tests every build in all, or most of the test files, but it definitely does take time and I don't see what benefit they provide.
Nothing that I think should hold up your `PR` getting merged though @danielgrijalva, just something for us to discuss as a team.

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-05-08T23:35:59Z" title="Thursday, May 9th 2019, 1:35:59 am +02:00">May 9, 2019</time>_

---


@mattwr18 sure, usually I start off by writing this kind of test case only to make sure I have everything set in place to start the component test. Later, I just don't remove it. I won't miss it so badly, so go ahead removing it.